aten/src/ATen/core/dispatch/OperatorEntry.h

```diff
@@ -81,32 +27,42 @@ class OperatorEntry final {
     return schema_;
   }
 
-  OpKernel lookupKernel(const Stack* stack) const {
-    return dispatchTable_.read([&] (const DispatchTable& dispatchTable) {
-      const DispatchTableEntry& kernel = dispatchTable.lookup(stack);
-      return OpKernel(kernel.kernel_func, kernel.cache_creator_func, kernel.unboxed_kernel_func);
+  template<class Return, class... Args>
+  Return callUnboxed(TensorTypeId dispatchKey, Args... args) const {
+    // TODO Remove dispatchKey argument and instead infer dispatchKey from args...
```

Comment:
As I think you are aware, `multi_dispatch_tensor_type_set` should work well for this.

Reply:
yes, I designed it towards being used with that.